# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/android/nsDragService.h
# Commit: 384b025f56d3
# Full Hash: 384b025f56d3f87e045ecede6b1a16b71c0b5d4f
# Author: David P <daparks@mozilla.com>
# Date: 2024-07-04 16:09:52
# Regressor Bug: 1893119
# File Overlap Count: 1
# Description:
#   Bug 1893119: Part 21 - Separate nsIDragService and nsIDragSession implementations r=gstoll,geckoview-reviewers,rkraesig,win-reviewers,m_kato
#   
#   Split the class inheritance trees for nsIDragService and nsIDragSession.
#   Remember that, before the start of this patch series, the inheritance
#   diagram was:
# ==============================================================================

diff -r beeda5ef3712 -r 384b025f56d3 widget/android/nsDragService.h
--- a/widget/android/nsDragService.h	Thu Jul 04 07:48:11 2024 +0000
+++ b/widget/android/nsDragService.h	Thu Jul 04 07:48:11 2024 +0000
@@ -13,15 +13,11 @@
 
 class nsITransferable;
 
-// Temporary inheritance from nsBaseDragService instead of nsBaseDragSession
-// (which nsBaseDragService temporarily inherits).
-// This will be undone at the end of this patch series.
-class nsDragSession : public nsBaseDragService {
+/**
+ * Android native nsIDragSession implementation
+ */
+class nsDragSession : public nsBaseDragSession {
  public:
-  nsDragSession() = default;
-
-  NS_DECL_ISUPPORTS_INHERITED
-
   // nsIDragSession
   NS_IMETHOD GetData(nsITransferable* aTransferable, uint32_t anItem) override;
   NS_IMETHOD GetNumDropItems(uint32_t* aNumItems) override;
@@ -32,6 +28,8 @@
 
   void SetData(nsITransferable* aTransferable);
 
+  void SetDropData(mozilla::java::GeckoDragAndDrop::DropData::Param aDropData);
+
   virtual bool MustUpdateDataTransfer(mozilla::EventMessage aMessage) override;
 
   MOZ_CAN_RUN_SCRIPT nsresult EndDragSessionImpl(
@@ -40,6 +38,12 @@
  protected:
   virtual ~nsDragSession() = default;
 
+  // nsBaseDragSession
+  MOZ_CAN_RUN_SCRIPT nsresult
+  InvokeDragSessionImpl(nsIWidget* aWidget, nsIArray* anArrayTransferables,
+                        const mozilla::Maybe<mozilla::CSSIntRegion>& aRegion,
+                        uint32_t aActionType) override;
+
   mozilla::java::sdk::Bitmap::LocalRef CreateDragImage(
       nsINode* aNode, const mozilla::Maybe<mozilla::CSSIntRegion>& aRegion);
 
@@ -47,26 +51,15 @@
   nsCOMPtr<nsITransferable> mTransferable;
 };
 
-// Temporary inheritance from nsDragSession instead of nsBaseDragService
-// (which nsDragSession temporarily inherits).
-// This will be undone at the end of this patch series.
-class nsDragService final : public nsDragSession {
+/**
+ * Android native nsIDragService implementation
+ */
+class nsDragService final : public nsBaseDragService {
  public:
   static already_AddRefed<nsDragService> GetInstance();
 
-  NS_DECL_ISUPPORTS_INHERITED
-
-  static void SetDropData(
-      mozilla::java::GeckoDragAndDrop::DropData::Param aDropData);
-
  protected:
-  virtual ~nsDragService() = default;
-
-  // nsBaseDragService
-  MOZ_CAN_RUN_SCRIPT nsresult
-  InvokeDragSessionImpl(nsIWidget* aWidget, nsIArray* anArrayTransferables,
-                        const mozilla::Maybe<mozilla::CSSIntRegion>& aRegion,
-                        uint32_t aActionType) override;
+  already_AddRefed<nsIDragSession> CreateDragSession() override;
 };
 
 #endif  // nsDragService_h__